lib/syntax_tree/parser.rb

```diff
@@ -1491,11 +1491,12 @@ def on_elsif(predicate, statements, consequent)
         ending.location.start_column
       )
 
-      Elsif.new(
+      IfNode.new(
         predicate: predicate,
         statements: statements,
         consequent: consequent,
-        location: beginning.location.to(ending.location)
+        location: beginning.location.to(ending.location),
+        beginning: beginning.value
```

Comment:
I made a couple choices here:
- I named this argument `beginning` because it seemed that other nodes also had an attribute with the same name, and I wanted to follow existing patterns.
- I decided to pass the value of the keyword as a string because there are other places where `IfNode`s are constructed manually, and I wanted to make it easy to create a value to pass in here.

Let me know if those both sound okay!

Reply:
`beginning` is fine as a name, but I think I've used `keyword` in other places.

I see what you mean about making it easy to be constructed, but I think just to keep consistency I'd prefer passing the whole keyword.